documentation/Templates.md

```diff
@@ -67,6 +67,8 @@ Additionally to the [tags and filters](https://stencil.fuller.li/en/latest/built
 | Node | Description |
 | ---- | ----------- |
 | `{% set my_capture_variable %}`<br>`{% endset %}` | node pair allowing you to capture the rendering of nodes inside it, so you can reuse it later by reusing `{{ my_capture_variable }}`.<br>This is useful if you intend to reuse the same content at multiple places — like the `sceneID` is for the `storyboards-default.stencil` template for example. |
+| `{% func my_block a b c %}`<br>`{{args.a}} - {{args.b}} - {{args.c}}`<br>`{% endfunc %}` | node pair allowing you to define a reusable block of nodes inside (like a function), so you can reuse it later.<br>In contrast to the `set` tag, the capturing for `func` is lazy, it will only be processed when actually called. When processed, the block will provide a variable `args` containing all the parameter values.<br>Similar to the `for` tag, this tag provides it's own context scope when called, so variables inside of it are not available outside of it. |
```

Comment:
Sure, I can remove it, makes the syntax (to use vars) shorter.

By the way, with this you can pass blocks around. They are actually variables in the context, so you could bass a block A to another block B (as a parameter) and within B call A.

Would prefer the Jinja2 syntax? Switching to `macro` instead of `func` is easy. The parameter/argument syntax is more difficult, currently I'm just using the functionality already in Stencil. The passing jinja blocks as parameters can be done using a combination of the `Set` tag and `call` tag.

Note: yeah, we also want to merge `set` upstream 😆